frontend/tests/components/search/SavedSearchQuery.test.tsx

```diff
@@ -0,0 +1,96 @@
+import { render, screen } from "@testing-library/react";
+import { useCopyToClipboard } from "src/hooks/useCopyToClipboard";
+import { useSnackbar } from "src/hooks/useSnackbar";
+
+import SavedSearchQuery from "src/components/search/SavedSearchQuery";
+
+const copyText = "copy this";
+const copyingText = "copying";
+const copiedText = "copied";
+const helpText = "click this to copy and share";
+const snackbarMessage = "you did it";
+
+jest.mock("src/hooks/useCopyToClipboard", () => ({
+  useCopyToClipboard: jest.fn(),
+}));
+
+jest.mock("src/hooks/useSnackbar", () => ({
+  useSnackbar: jest.fn(),
+}));
+
+jest.mock("src/components/TooltipWrapper", () => {
+  return {
+    __esModule: true,
+    default: () => {
+      return <div>{copyText}</div>;
+    },
+  };
+});
+
+const mockedUseCopyToClipboard = useCopyToClipboard as jest.MockedFunction<
```

Comment:
I'm confused about how these mocks are working. Would it work to set up the mocks and `jest.fn`s and reference them in the mocked modules, rather than importing the real modules?

Reply:
Yeah, that's exactly the sort of implementation I was thinking of. If I throw this in, all the tests still pass for me locally. Were that not passing for you?

```suggestion
jest.mock("src/hooks/useCopyToClipboard", () => ({
  useCopyToClipboard: () => mockedUseCopyToClipboard(),
}));

jest.mock("src/hooks/useSnackbar", () => ({
  useSnackbar: () => mockedUseSnackBar(),
}));

const mockedUseCopyToClipboard = jest.fn();
const mockedUseSnackBar = jest.fn();
```

Reply:
When testing your suggestion, there is a linting error:

```
./tests/components/search/SearchQueryCopyButton.test.tsx
24:29  Error: Unsafe return of an `any` typed value.  @typescript-eslint/no-unsafe-return
27:22  Error: Unsafe return of an `any` typed value.  @typescript-eslint/no-unsafe-return
```

which is fixed by:

```javascript

const mockedUseCopyToClipboard = jest.fn() as jest.MockedFunction<
  typeof useCopyToClipboard
>;

```

which is I guess is slightly better because it only uses the functions as types, but they are still mocked so not sure what that gains us 🤷🏿 . I updated it as I had tested it locally anyway.

Reply:
@doug-s-nava additional to the linting, when I use your suggestion I get:

```
  ● Test suite failed to run

    ReferenceError: Cannot access 'mockedUseCopyToClipboard' before initialization

      21 |
      22 | jest.mock("src/hooks/useCopyToClipboard", () => ({
    > 23 |   useCopyToClipboard: mockedUseCopyToClipboard,
         |                       ^
      24 | }));
      25 |
```